common/server_protocolhandler.cpp

```diff
@@ -373,6 +383,20 @@ void Server_ProtocolHandler::pingClockTimeout()
 
     if (timeRunning - lastDataReceived > server->getMaxPlayerInactivityTime())
         prepareDestroy();
+
+    if (QString::fromStdString(userInfo->privlevel()).toLower() == "none")
+        if ((timeRunning - lastActionReceived) == round(server->getIdleClientTimout() *.9)) {
```

Comment:
Nice idea to add a warning, but i see a problem here.
This will only work if the config value of `clientkeepalive` is set to the default of 1, so the `pingClockTimeout()` method is called every second.
If a server admin sets that value to eg. 5 secs, there's only 20% probability that `pingClockTimeout()` gets executed exactly `server->getIdleClientTimout() *.9` seconds after the last action has been received.

A quick workaround could be to introduce the value of `clientkeepalive` in the equation, eg:
```
int pingClockInterval = settingsCache->value("server/clientkeepalive", 1).toInt();
int delta = timeRunning - lastActionReceived - round(server->getIdleClientTimout() *.9);
if(delta > 0 && delta < clientkeepalive) { ... }
```
I'm not exactly a fan of this solution, if you find something better go for it.